### DISCUSSION TITLE
Unable to list or recover from volume

### DISCUSSION THREAD

**Discussion by matthewcmead** [jivanpal/apfs-tools#47] [Q&A]
I have a volume in which /Users/<username> has disappeared.  I can search the volume using the search command (recompiled to have a list of names I'm interested in) and find various matches which make me suspect the data is still in the volume.

When I attempt to list anything on the volume by fsoid (found from search) or by path (e.g., /Users/<username>) I get the following:

- Volume 0 has Virtual OID 0x6400000000 ... END: No objects with Virtual OID 0x6400000000 and maximum XID 0x915 exist in `nx_omap_btree`.

Any ideas on how to get to the point I can list / recover from this volume?

Thank you!

**Comment by jivanpal** [comment]
This means that the object map of the latest transaction doesn't contain any records for the object you're interested in, and you'll have to (1) explore an older transaction and/or (2) scan the entire disk for object map records that you're interested in. These are tasks which Drat currently isn't capable of (hence me closing this thread), though I would like to add these features.

For now, you may be interested in alternative software like Disk Drill, which is capable of (2).
